Mathlib/Algebra/Module/Injective.lean

```diff
@@ -446,32 +442,105 @@ theorem extensionOfMax_to_submodule_eq_top (h : Module.Baer R Q) :
+    LinearMap.ext fun x ↦ ((extensionOfMax f g).is_extension x).symm
+
 /-- **Baer's criterion** for injective module : a Baer module is an injective module, i.e. if every
 linear map from an ideal can be extended, then the module is injective.-/
 protected theorem injective (h : Module.Baer R Q) : Module.Injective R Q :=
-  { out := fun X Y ins1 ins2 ins3 ins4 i hi f =>
-      haveI : Fact (Function.Injective i) := ⟨hi⟩
-      ⟨{  toFun := fun y =>
-            (extensionOfMax i f).toLinearPMap
-              ⟨y, (extensionOfMax_to_submodule_eq_top i f h).symm ▸ trivial⟩
-          map_add' := fun x y => by
-            rw [← LinearPMap.map_add]
-            congr
-          map_smul' := fun r x => by
-            rw [← LinearPMap.map_smul]
-            -- Porting note: used to be congr
-            dsimp },
-        fun x => ((extensionOfMax i f).is_extension x).symm⟩ }
+  { out := fun X Y ins1 ins2 ins3 ins4 i hi f ↦ by
+      obtain ⟨h, H⟩ := Module.Baer.extension_property h i hi f
+      exact ⟨h, FunLike.congr_fun H⟩ }
 set_option linter.uppercaseLean3 false in
 #align module.Baer.injective Module.Baer.injective
 
-protected theorem of_injective (inj : Module.Injective R Q) : Module.Baer R Q := fun I g ↦
-  let ⟨g', hg'⟩ := inj.1 (ULift.{max u v} I) (ULift.{max u v} R)
-    (ULift.moduleEquiv.symm.toLinearMap ∘ₗ I.subtype ∘ₗ ULift.moduleEquiv.toLinearMap)
-    (fun a b h ↦ by aesop) (g ∘ₗ ULift.moduleEquiv.toLinearMap)
-  ⟨g' ∘ₗ ULift.moduleEquiv.symm.toLinearMap, by aesop⟩
+protected theorem of_injective [UnivLE.{u, v}] (inj : Module.Injective R Q) : Module.Baer R Q := by
+  intro I g
+  let eI := Shrink.linearEquiv I R
+  let eR := Shrink.linearEquiv R R
 
-protected theorem iff_injective : Module.Baer R Q ↔ Module.Injective R Q :=
-  ⟨Module.Baer.injective, Module.Baer.of_injective⟩
+  obtain ⟨g', hg'⟩ := Module.Injective.out (eR.symm.toLinearMap ∘ₗ I.subtype ∘ₗ eI.toLinearMap)
+    (eR.symm.injective.comp <| Subtype.val_injective.comp eI.injective) (g ∘ₗ eI.toLinearMap)
+
+  exact ⟨g' ∘ₗ eR.symm.toLinearMap, fun x mx ↦ by simpa using hg' (equivShrink I ⟨x, mx⟩)⟩
+
+protected theorem iff_injective [UnivLE.{u, v}] : Module.Baer R Q ↔ Module.Injective R Q :=
+  ⟨Module.Baer.injective, Module.Baer.of_injective.{u, v}⟩
 
 end Module.Baer
+
+section ULift
+
+variable {M : Type v} [AddCommGroup M] [Module R M] [UnivLE.{u, v}]
+
+lemma Module.ulift_injective_of_injective
+    (inj : Module.Injective R M) :
+    Module.Injective R (ULift.{max v' v} M) := by
+  letI : UnivLE.{u, max v' v} := UnivLE.trans.{v, u, max v' v}
+  rw [← Module.Baer.iff_injective.{u, v}] at inj
+  rw [← Module.Baer.iff_injective.{u, max v' v}]
+  intro I g
+  obtain ⟨g', hg'⟩ := inj I (ULift.moduleEquiv.toLinearMap ∘ₗ g)
+  exact ⟨ULift.moduleEquiv.symm.toLinearMap ∘ₗ g', fun r hr ↦ ULift.ext _ _ <| hg' r hr⟩
+
+lemma Module.injective_of_ulift_injective
+    (inj : Module.Injective R (ULift.{max v v'} M)) :
+    Module.Injective R M := by
```

Comment:
These should say `Ulift.{v}` instead of `ULift.{max v v'}`, and the second doesn't require UnivLE. See [my branch](https://github.com/leanprover-community/mathlib4/compare/zjj/liftInjectiveModules...jyxu/liftInjectiveModules?expand=1).